### PR TITLE
[SelectionDAG] Rename SDNode::uses() to users().

### PR DESCRIPTION
**Pull request by topperc**
This function is most often used in range based loops or algorithms where the iterator is implicitly dereferenced. The dereference returns an SDNode * of the user rather than SDUse * so users() is a better name.

I've long beeen annoyed that we can't write a range based loop over SDUse when we need getOperandNo. I plan to rename use_iterator to user_iterator and add a use_iterator that returns SDUse* on dereference. This will make it more like IR.

### CODE REVIEW

File: llvm/include/llvm/CodeGen/SelectionDAGNodes.h [hunk 1]
```diff
@@ -750,7 +750,7 @@ END_TWO_BYTE_PACK()
   bool use_empty() const { return UseList == nullptr; }
 
   /// Return true if there is exactly one use of this node.
-  bool hasOneUse() const { return hasSingleElement(uses()); }
+  bool hasOneUse() const { return hasSingleElement(users()); }
```
Comment (s-barannikov) [inline llvm/include/llvm/CodeGen/SelectionDAGNodes.h:753]: Changing this name will be difficult, but maybe it is worth fixing the comment at least?
This method returns one if all users are the same node.

**Review by s-barannikov — APPROVED** [review]
LGTM with a tiny nit.

File: llvm/include/llvm/CodeGen/SelectionDAGNodes.h (hunk 1 above)
Comment (s-barannikov) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h:753]: It looks like I'm wrong. I had a strong feeling that this returns true if all uses are the same node, but after looking around it does not appear so. Probably mistaken for some other method. Sorry.

Comment (topperc) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h:753]: >This method returns true if all users are the same node.

That's not true. hasSingleElement just checks that the range contains 1 element. We don't do anything to remove duplicate users from the range.

Comment (s-barannikov) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h:753]: Yep, sorry.

**Review by arsenm — APPROVED** [review]

**Pull request merged**